x-pack/plugins/lens/public/trigger_actions/open_lens_config/create_action_helpers.ts

```diff
@@ -31,20 +32,41 @@ export const [getDatasourceMap, setDatasourceMap] = createGetterSetter<
   Record<string, Datasource<unknown, unknown>>
 >('DatasourceMap', false);
 
-export function isCreateActionCompatible(core: CoreStart) {
-  return core.uiSettings.get(ENABLE_ESQL);
+export async function isCreateActionCompatible(
+  core: CoreStart,
+  editorFrameService: EditorFrameService
+) {
+  let isCompatible: boolean = core.uiSettings.get(ENABLE_ESQL);
+
+  if (isCompatible && !getDatasourceMap() && !getVisualizationMap()) {
```

Comment:
Shouldn't this be:
`if (isCompatible && (!getDatasourceMap() || !getVisualizationMap())) {`
?

Reply:
Thanks for the catch!! You're right having `||` would be much better